### PR TITLE
Add rclone to Vsaio

### PR DESCRIPTION
**Pull request by matthewoliver**
Automatically install rclone and setup rclone.conf with all the tempauth accounts setup for both swift and s3.

### CODE REVIEW

File: cookbooks/swift/recipes/setup.rb [hunk 1]
```diff
@@ -239,6 +239,30 @@
...
+  owner node['username']
+  group node['username']
+  mode 0700
+  action :create
+end
+directory "/home/#{node['username']}/.config/rclone" do
+  owner node['username']
+  group node['username']
+  mode 0700
+  action :create
```
Comment (tipabu) [inline cookbooks/swift/recipes/setup.rb:254]: nit: Could add a `recursive true` here and skip the parent dir above.

File: cookbooks/swift/recipes/setup.rb [hunk 2]
```diff
@@ -239,6 +239,30 @@
...
+end
+template "/home/#{node['username']}/.config/rclone/rclone.conf" do
+  source "/home/rclone/rclone.erb"
+  owner node['username']
+  group node['username']
+  mode 0700
+  variables({
+    :ssl => node['ssl'],
+    :hostname => node['hostname'],
```
Comment (tipabu) [inline cookbooks/swift/recipes/setup.rb:263]: Might want to use
```
    :base_uri => node['base_uri'],
    :auth_uri => node['auth_uri'],
```
instead, just to minimize the places where we have to know how to reconstruct them.

File: cookbooks/swift/templates/default/home/rclone/rclone.erb [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+type = swift
+env_auth = false
+user = admin:admin
+key = admin
+auth = <%= @ssl ? "https://#{@hostname}" : "http://#{@hostname}:8080" %>/auth/v1.0
+auth_version = 1
+
+[admin-s3]
+type = s3
+env_auth = true
```
Comment (tipabu) [inline cookbooks/swift/templates/default/home/rclone/rclone.erb:11]: Why `env_auth = true` for S3? We're explicitly setting creds in the file, do we really want to allow the environment to override?

**Review by tipabu — COMMENTED** [review]
Hey, I'd been meaning to do this, too! Good idea to add all the profiles. One thought about what variables to plumb in, and one question about `env_auth`.

File: cookbooks/swift/recipes/setup.rb (hunk 1 above)
Comment (matthewoliver) [reply cookbooks/swift/recipes/setup.rb:254]: great idea! My chef-foo is a little rusty :)

File: cookbooks/swift/recipes/setup.rb (hunk 2 above)
Comment (matthewoliver) [reply cookbooks/swift/recipes/setup.rb:263]: ahh, yeah, much better!
Mental note, check all the vars before trying to build your own ;)

File: cookbooks/swift/templates/default/home/rclone/rclone.erb (hunk 3 above)
Comment (matthewoliver) [reply cookbooks/swift/templates/default/home/rclone/rclone.erb:11]: I stole this from the autogenerate ones given to us via our downstream portal... so no idea :P I'll change it.

**Pull request merged**